include/fixed_containers/optional_ref.hpp

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+#include "fixed_containers/concepts.hpp"
+#include "fixed_containers/optional_ref_checking.hpp"
+
+#include <compare>
+#include <functional>
+#include <optional>
+
+namespace fixed_containers
+{
+/*
+ * OptionalRef<T> is intended to mimic std::optional<std::reference_wrapper<T>>
+ * but without the extra indirection when dereferencing.
+ */
+template <IsNotReference T,
+          customize::OptionalRefChecking<T> CheckingType = customize::OptionalRefAbortChecking<T>>
+class OptionalRef
+{
+private:
+    using Self = OptionalRef<T, CheckingType>;
+    using ref = T&;
+    using ptr = T*;
```

Comment:
Is there a reason why we're using the pointer type as the backing type and not simply the reference type?